packages/core/src/loading/stream_loader.ts

```diff
@@ -0,0 +1,75 @@
+import { LogSyncer } from './log_syncer.js'
+import { TipFetcher } from './tip_fetcher.js'
+import { StateManipulator } from './state_manipulator.js'
+import {
+  AnchorValidator,
+  DiagnosticsLogger,
+  StreamState,
+  StreamUtils,
+} from '@ceramicnetwork/common'
+import { StreamID } from '@ceramicnetwork/streamid'
+import { Dispatcher } from '../dispatcher.js'
+import { verifyAnchorAndApplyTimestamps } from '../conflict-resolution.js'
+
+/**
+ * Class to contain all the logic for loading a stream, including fetching the relevant commit
+ * data from the p2p network and applying the commits to get a StreamState.  It notably however does
+ * not manage the persistence of the state information, nor updating the cache or indexing. It is
+ * purely stateless.
+ */
+export class StreamLoader {
```

Comment:
Nice! I was thinking about adding such module to avoid using the repository in many places that only need to load streams. Would this module be safe to use around and expose in the context without creating circular dependencies?

Also, what do you think about adding the caching logic there please? If we don't, isn't there a risk we'll have duplicate attempts to load the same streams?

Reply:
As we discussed this morning, my goal is to leave the StreamLoader as purely functional and stateless, and have higher-level modules (currently the Repository) be responsible for things like caching, updating the state store, updating the index, and concurrency control